modules/navier_stokes/src/postprocessors/PressureDrop.C

```diff
@@ -0,0 +1,339 @@
+    {
+      if (std::find(_upstream_boundaries.begin(), _upstream_boundaries.end(), bdy) !=
+          _upstream_boundaries.end())
+      {
+        upstream = true;
+        status_known = true;
+#ifdef NDEBUG
+        break;
+#else
+        if (std::find(_downstream_boundaries.begin(), _downstream_boundaries.end(), bdy) !=
+            _downstream_boundaries.end())
+          mooseError("Boundary '", _mesh.getBoundaryName(bdy), "' is both upstream and downstream");
+#endif
+      }
+#ifndef NDEBUG
+      if (upstream &&
+          std::find(_downstream_boundaries.begin(), _downstream_boundaries.end(), bdy) !=
+              _downstream_boundaries.end())
+        mooseError("Face info is part of both upstream and downstream boundaries");
+#endif
+      if (!upstream &&
+          std::find(_downstream_boundaries.begin(), _downstream_boundaries.end(), bdy) !=
+              _downstream_boundaries.end())
+        status_known = true;
+
+        // in debug mode we will check all boundaries the face info is a part of
+        // to make sure they are consistently upstream or downstream
+#ifdef NDEBUG
+      if (status_known)
+        break;
+#endif
+    }
+    // we ll assume all face infos for a given element and side have the same upstream status
+    // it seems very unlikely that upstream and downstream boundaries would be touching
+    // and the delimitation would be within a refined element's face
+    if (status_known)
+      break;
+  }
+
+  if (upstream)
+  {
+    // Integration loops are different for FE and FV at this point
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_upstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_upstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_upstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_upstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+  // Downstream contributions
+  else
+  {
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_downstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_downstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_downstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_downstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+}
+
+Real
+PressureDrop::computeFaceInfoWeightedPressureIntegral(const FaceInfo * const fi) const
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
```

Comment:
Will have to monitor #23665 to tell whether to keep this

Reply:
ok will remove